test/Microsoft.Azure.WebJobs.Extensions.Kafka.UnitTests/KafkaListenerTest.cs

```diff
@@ -156,6 +159,62 @@ public async Task When_Using_MultiItem_Binding_10_Events_Should_Execute_Function
             executor.Verify(x => x.TryExecuteAsync(It.IsNotNull<TriggeredFunctionData>(), It.IsAny<CancellationToken>()), Times.Once);
         }
 
+        [Fact]
+        public void ScaleMonitor_Id_ReturnsExpectedValue()
```

Comment:
Once you refactor as I suggested above so that KafkaTopicScaler implements IScaleMonitor, these tests will be easier to write because you don't have to mock out all the listener stuff. You then only test the GetMonitor method in this test suite, and add a new test suite for KafkaTopicScaler.